### DISCUSSION TITLE
GPU Reset Recovery

### DISCUSSION THREAD

**Discussion by travankor** [niri-wm/niri#700] [Ideas]
On KDE and wlroots compositors, the compositor will try to recover the session if the GPU driver crashes.

Can niri do the same with smithay?

**Comment by YaLTeR** [comment]
I don't think smithay has the necessary parts implemented. Might be mistaken.

**Comment by Lyndeno** [comment]
With recent news about GNOME making progress on this: I'd like to revive this conversation. https://blogs.gnome.org/anonymoux47/2026/07/02/gpu-reset-recovery-in-mutter-a-progress-update/

This would be useful to have.

**Reply by YaLTeR** [reply]
I mean, there's nothing to revive here. My comment still applies
